src/woo_search/search_index/api/serializers/search.py

```diff
@@ -0,0 +1,74 @@
+from django.utils.translation import gettext_lazy as _
+
+from drf_polymorphic.serializers import PolymorphicSerializer
+from rest_framework import serializers
+
+from ...constants import ResultTypeChoices, SortChoices
+from . import DocumentSerializer, PublicationSerializer
+
+
+class DocumentPolymorphicSerializer(serializers.Serializer):
+    record = DocumentSerializer(read_only=True)
+
+
+class PublicationPolymorphicSerializer(serializers.Serializer):
+    record = PublicationSerializer(read_only=True)
+
+
+class SearchSerializer(serializers.Serializer):
+    query = serializers.CharField(
+        required=False,
+        help_text=_(
+            "Filtering records based on the provided search term."
+            " This field searches the data within the following fields"
+            " (with priority based on the order below):\n\n"
+            "- `identifier` **field only present in Document*\n"
+            "- `officieleTitel`\n"
+            "- `verkorteTitel`\n"
+            "- `omschrijving`"
+        ),
+    )
+    page = serializers.IntegerField(
+        default=1, help_text=_("A page number within the paginated result set.")
+    )
+    page_size = serializers.IntegerField(
+        default=10, help_text=_("Number of results to return per page.")
+    )
+    sort = serializers.ChoiceField(
+        choices=SortChoices.choices,
+        default=SortChoices.relevance,
+    )
+    result_type = serializers.ChoiceField(
+        choices=ResultTypeChoices.choices,
+        default=ResultTypeChoices.all,
+        help_text=_("Which table(s) should be present in the returning body."),
+    )
+
+    def to_internal_value(self, data):
+        data = super().to_internal_value(data)
+        page = data["page"]
+        page_size = data["page_size"]
+
+        # Elasticsearch's pagination doesn't work with pages.
+        # It instead has a `from` argument which states from which record
+        # it will return values.
+        data["page"] = page_size * (page - 1)
```

Comment:
After overwriting `page` the value of it no longer makes sense in the context of `page` + `page_size`, so maybe we could add a read only field `from` to put this value in?

Reply:
I'm wondering if it wouldn't make more sense to do this conversion in the view rather than the serializer. Imagine that you'd want to save a search query in the database as "pre-defined" searches. This would mangle the `page` parameter and executing it again would lead to different results.

That would address the comments from Steven as well, because we're no longer mutating stuff. Mutation in the context of serializing/deserializing is an anti-pattern anyway, so pushing this down to the view seems more appropriate.